core/src/utils/test/utils.ts

```diff
@@ -185,3 +185,20 @@ export const checkModeClasses = async (el: E2EElement, globalMode: string) => {
   const mode = (await el.getProperty('mode')) || globalMode;
   expect(el).toHaveClass(`${mode}`);
 };
+
+/**
+ * Wait for the browser to fire an event (includes custom events).
+ * @param page The Puppeteer page object
+ * @param event The name of the event.
+ * @param timeout The amount of time to wait for the event to fire (defaults 5000ms)
+ * @returns Resolves when the event fires or the timeout is reached.
+ */
+export const waitForEvent = async (page: E2EPage, event: string, timeout = 5000) => {
+  return Promise.race([
+    page.evaluate(
+      event => new Promise(resolve => document.addEventListener(event, resolve, { once: true })),
+      event
+    ),
```

Comment:
This feature already exists in the Stencil/Puppeteer testing library. You can do `const ev = await page.spyOnEvent(...);` and then do `await ev.next();`.

Reply:
Also it looks like `listenForEvent` is redundant too if we use `spyOnEvent`

Reply:
Ahh... that works flawlessly. Thank you! I created a task in our tech debt epic to update existing `listenForEvent` usages over to `spyOnEvent`.